stdlib/public/core/ContiguousArrayBuffer.swift

```diff
@@ -256,7 +256,7 @@ public struct _ContiguousArrayBuffer<Element> : _ArrayBufferProtocol {
 
   /// If the elements are stored contiguously, a pointer to the first
   /// element. Otherwise, `nil`.
-  public var firstElementAddress: UnsafeMutablePointer<Element> {
+  public var firstElementAddress: UnsafeMutablePointer<Element>? {
```

Comment:
It might be better to define `firstElementAddress` in terms of a non-optional internal property to satisfy the conformance while allowing the implementation to directly use the non-optional property.